src/Mod/Fem/femobjects/material_common.py

```diff
@@ -97,9 +100,48 @@ def onDocumentRestored(self, obj):
                 # change References to App::PropertyLinkSubListGlobal
                 prop.handle_change_type(obj, old_type="App::PropertyLinkSubList")
 
+        # try update UUID from Material
+        if not obj.UUID:
+            obj.UUID = self._get_material_uuid(obj.Material, obj.UUID)
+
         if not obj.hasExtension("App::SuppressibleExtensionPython"):
             obj.addExtension("App::SuppressibleExtensionPython")
 
+    def _get_material_uuid(self, material, uuid):
+        material_manager = Materials.MaterialManager()
+        if uuid:
+            try:
+                material_manager.getMaterial(uuid)
+                return uuid
+            except:
```

Comment:
The linter is right here, we should catch a more specific exception. What does `getMaterial` raise?

Reply:
I have edited the function and the try statement was removed.